services/serviceutils/src/main/java/tech/pegasys/artemis/service/serviceutils/Service.java

```diff
@@ -47,4 +47,8 @@
   }
 
   protected abstract SafeFuture<?> doStop();
+
+  public State getState() {
```

Comment:
If you want to test the `state` transitions, suggest making this package-private, marking it `@VisibleForTesting` and adding a `ServiceTest` rather than checking the transitions in child classes.

Reply:
no longer needed as I removed the genericized Service class